types/items/item-history.d.ts

```diff
@@ -1,4 +1,23 @@
 export = ItemHistory;
+
+/**
+ * Item state with timestamp
+ */
+export type HistoricItem = {
```

Comment:
Instead of directly defining the `HistoricItem` type, rather defined it in JSDoc at the top of the item-history.js file (after the imports), because JSDoc needs some type definition as well and unfortunantely does not read the .d.ts files.

```javascript
/**
 * @typedef {object} HistoricItem
 * @property {string} state Item state
 * @property {HostState} rawState Raw Java state
 * @property {time.ZonedDateTime} timestamp timestamp of historic Item
 */
```

Reply:
I don't fully understand yet. Do you mean your suggested change *replaces* the item-history.d.ts one, or should this be added *in addition*?

Reply:
Have you written the HistoricItem in the `item-history.d.ts` file yourself?

Reply:
Yes, that's why it's in the diff ;-)

Reply:
Done.